clients/unshaded/src/main/java/tachyon/client/FileInStream.java

```diff
@@ -121,7 +123,8 @@ public int read(byte[] b, int off, int len) throws IOException {
     } else if (len == 0) {
       return 0;
     } else if (mCurrentPosition >= mFileLength) {
-      return -1;
+      throw new EOFException(String.format("Read position %s is past file length %s.",
```

Comment:
NIT for readability: maybe exchange `if (mCurrentPosition >= mFileLength)` branch with `if (len == 0)` branch?
so it looks like:
```else if (mCurrentPosition >= mFileLength)  {
  throw EOFException();
} else if (len == 0) {
  return 0;
}```

In this way, those Exceptions cases are thrown first